File: android/app/src/main/java/chat/rocket/reactnative/Ejson.java
package chat.rocket.reactnative;

public class Ejson {
    String host;
    String rid;
    String type;
    Sender sender;

    private String TOKEN_KEY = "reactnativemeteor_usertoken-";

    public String getAvatarUri() {
        if (type == null) {
            return null;
        }
        return serverURL() + "/avatar/" + this.sender.username;
    }

    public String serverURL() {
        String url = this.host;
        if (url != null && url.endsWith("/")) {
            url = url.substring(0, url.length() - 1);
        }
        return url;
    }

    public class Sender {
        String username;
        String _id;
    }
}

File: android/app/src/main/java/chat/rocket/reactnative/pushnotification/Ejson.kt
package chat.rocket.reactnative.pushnotification

import kotlinx.serialization.Serializable

@Serializable
data class Ejson(val host: String, val rid: String, val sender: Sender) {
    override fun toString(): String = host + rid

    val avatarUri: String = "${removeTrailingSlash(host)}/avatar/${sender.username}"

    private fun removeTrailingSlash(baseUrl: String): String {
        var url = baseUrl
        if (url.endsWith('/')) url = url.dropLast(1)
        return url
    }
}

@Serializable
data class Sender(val username: String)